src/Utilities/for_testing/CMakeLists.txt

```diff
@@ -8,6 +8,11 @@
 #//////////////////////////////////////////////////////////////////////////////////////
 
 SET(test_utilities_src
-  RandomForTest.cpp)
+  RandomForTest.cpp
+  checkMatrix.cpp)
 add_library(utilities_for_test ${test_utilities_src})
-TARGET_INCLUDE_DIRECTORIES(utilities_for_test PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
+target_include_directories(utilities_for_test PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
+target_include_directories(utilities_for_test PRIVATE "${PROJECT_SOURCE_DIR}/external_codes/catch")
+target_link_libraries(utilities_for_test PUBLIC containers)
+
+subdirs(tests)
```

Comment:
prefer `Utilities/tests/for_testing` to  `Utilities/for_testing/tests`

Reply:
That seems like it would double the work of modifying module directories layouts.
But I did notice the extra directory traversals in platform. 
Is the idea to have fewer test program targets?

Reply:
I'm thinking of have one target for each folder under `src`.
`src/Utilities/test` only needs to depend on qmcutils. So we don't need to care about to many fine-grain CMake targets.
Currently qmcutils is in at "src/CMakeLists.txt" but it should be moved to "src/Utilities/CMakeLists.txt" but this is an on-going cmake refactoring out of the scope of this PR.

Reply:
Ok I moved the tests.  I still prefer Utilities/for_testing as a separate target. They are not routines for use in the application code.

Reply:
That is OK.